draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,584 @@ and RTTVar are updated by the socket based on algorithms for the sender
+
+g. Stop;
+
+where
 
-For file transfer, any known File Congestion Control algorithms like CUBIC {{RFC8312}} and BBR {{BBR}} can apply, including the congestion control mechanism proposed in UDT {{GHG04b}}, {{GuAnAO}}. The UDT congestion control relies on the available link capacity, packet loss reports (NAK) and packet acknowledgements (ACKs). It then slows down the output of packets as needed by adjusting the packet sending pace. In periods of congestion, it can block the main stream and focus on the lost packets.
+- AvgNAKNum is the average number of NAKs during a congestion period.
+  Initial value: 0;
+- NAKCount is the number of NAKs received so far in the current
+  congestion period. Initial value: 0;
+- DecCount means the number of times that the sending rate has been
+  decreased during the congestion period. Initial value: 0;
+- DecRandom is a random number used to decide if the rate should be
+  decreased or not for the following NAKs (not the first one) during
+  the congestion period. DecRandom is a random number between 1 and
+  the average number of NAKs per congestion period (AvgNAKNum).
+
+Congestion period is defined as the time between two NAKs in which
+the biggest lost packet sequence number carried in the NAK is greater
+than the LastDecSeq.
+
+A slight modification of coefficients used in the above mentioned
+formulas has been made in order to reduce the value of sending
+rate decrease.
+
+Step 4. If DecCount <= 5, and NAKCount == DecCount * DecRandom:
+
+a. Update SND period: SND = SND * 1.03;
+
+b. Increase DecCount and NAKCount by 1;
+
+c. Record the current largest sent sequence number (LastDecSeq).
+
+#### Link Capacity and Receiving Rate Estimation {#bandwidth-estimation}
+
+Link capacity and receiving rate estimations, in packets/bytes per second,
+are calculated at the receiver side during the file transmission.
+It's worth noting that receiving rate estimation is used only during
+the slow start period of the file congestion control algorithm
+({{default-fileCC-slow-start}}), however it's available during
+the whole data transmission period.
+The latest estimation obtained before the end of the slow start
+period is used by the sender as a reference maximum speed
+to continue data transmission without further congestion.
+Link capacity is estimated all the time and used primarily
+(as well as packet loss ratio and other protocol statistics)
+for smoothed sending rate adjustments during the transmission process.
+
+The receiver records the inter-arrival time of each packet
+(time delta with the previous data packet) which is further
+used in the models to estimate bandwidth and receiving speed.
+The communication between receiver and sender happens by means
+of acknowledgment packets which are sent regularly
+(each 10 milliseconds) and contain some control information
+as well as bandwidth and delivery rate estimations.
+At the sender side, upon receiving a new value, a smoothed average
```

Comment:
smoothed?

Reply:
the same explanation as for RTT, receiving rate and link capacity reported by receiver via ACK packets
upon ACK arrival, at the sender side we apply EWMA one more time to all these estimates